src/Core/Settings.h

```diff
@@ -399,6 +399,9 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, max_network_bandwidth_for_user, 0, "The maximum speed of data exchange over the network in bytes per second for all concurrently running user queries. Zero means unlimited.", 0)\
     M(UInt64, max_network_bandwidth_for_all_users, 0, "The maximum speed of data exchange over the network in bytes per second for all concurrently running queries. Zero means unlimited.", 0) \
     \
+    M(UInt64, max_temp_data_on_disk_size_for_user, 0, "The maximum amount of data consumed by temporary files on disk in bytes for all concurrently running user queries. Zero means unlimited.", 0)\
```

Comment:
temp -> temporary please.

Reply:
Need "temporary" instead of "temp" for the following reasons:
1. We hate abbrs wholeheartedly.
2. If you abbreviate "temporary", it is difficult to remember if it should be "temp" or "tmp" or "t".
3. There are other settings with the full "temporary" word.
4. The "temp" abbreviation can be confused with the word "tempo".

Reply:
Let's rename it before it is in release https://github.com/ClickHouse/ClickHouse/pull/41984

Also, I noticed that there's no setting for global. I'll add it. Where should it be configured? In config.xml, I suppose, not in settings? What section?

Reply:
config.xml, similarly to `max_memory_usage_for_server`.